improve handling of error pages

Andtalath opened the pull request:
the current version can handle files missing, but it can not handle directories missing.

https://docs.stakater.com/mto/main/reference-guides/admin-clusterrole.html Fails to show a nice looking 404.

https://docs.stakater.com/mto/main/obviously_missing_file succeeds.

With this change, both should work.

Not sure what internal does or if is needed, have tested on a local nginx only.

stakater-user commented:
@rasheedamir Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:SNAPSHOT-PR-146-263e3ebb`

abdulhaseeb2 reviewed (approved):
LGTM

stakater-user commented:
@abdulhaseeb2 Image is available for testing. `docker pull docker-openshift-stakater-nexus.apps.prod.clusters.stakater.io/stakater/StakaterCloudDocs:v0.0.93`

Andtalath committed:
improve handling of error pages

the current version can handle files missing, but it can not handle
directories missing.

https://docs.stakater.com/mto/main/reference-guides/admin-clusterrole.html
Fails to show a nice looking 404.

https://docs.stakater.com/mto/main/obviously_missing_file
succeeds.

With this change, both should work.

Not sure what internal does or if is needed, have tested on a local
nginx only.
